### PR TITLE
refactor:  Deploy by cloning

### PR DESCRIPTION
**Pull request by xBA5ED**
# Description

*What does this PR: do, how, why?*

## Limitations & risks

*Are there any trade-off or new vulnarbility surface based on theses changes?*

# Check-list
- [ ] Tests are covering the new feature
- [ ] Code is [natspec'd](https://docs.soliditylang.org/en/v0.8.17/natspec-format.html)
- [ ] Code is [linted and formatted](https://docs.soliditylang.org/en/v0.8.17/style-guide.html)
- [ ] I have run the test locally (and they pass)
- [ ] I have rebased to the latest main commit (and tests still pass)

# Interactions
These changes will impact the following contracts:
- Directly:

- Indirectly:

### CODE REVIEW

File: src/interfaces/IJBSuckerDeployer.sol [hunk 1]
```diff
@@ -7,12 +7,15 @@ import {IJBTokens} from "@bananapus/core/src/interfaces/IJBTokens.sol";
 import {IJBSucker} from "./IJBSucker.sol";
 
 interface IJBSuckerDeployer {
+    error JBSuckerDeployer_Unauthorized(address caller, address expected);
```
Comment (mejango) [inline src/interfaces/IJBSuckerDeployer.sol]: nit: alphabetize error definitions

File: src/interfaces/IJBCCIPSuckerDeployer.sol [hunk 2]
```diff
@@ -2,13 +2,14 @@
...
     function createForSender(uint256 localProjectId, bytes32 salt) external returns (IJBSucker sucker);
 
-    function TEMP_ID_STORE() external view returns (uint256);
+    function remoteChainId() external view returns (uint256);
 
-    function REMOTE_CHAIN_ID() external view returns (uint256);
+    function remoteChainSelector() external view returns (uint64);
 
-    function REMOTE_CHAIN_SELECTOR() external view returns (uint64);
+    function ccipRouter() external view returns (ICCIPRouter);
```
Comment (mejango) [inline src/interfaces/IJBCCIPSuckerDeployer.sol]: nit: alphebetize function defs

File: src/JBSucker.sol [hunk 3]
```diff
@@ -117,6 +112,12 @@ abstract contract JBSucker is JBPermissioned, ERC165, IJBSucker {
     // -------------------- internal stored properties ------------------- //
     //*********************************************************************//
 
+    /// @notice The peer sucker on the remote chain.
+    address private remotePeer;
+
+    /// @notice The ID of the project (on the local chain) that this sucker is associated with.
+    uint256 private localProjectId;
```
Comment (mejango) [inline src/JBSucker.sol:119]: nit: alphebetize prop. 
why "private" and not "internal"?

**Review by mejango — APPROVED** [review]
just some dumb nits

File: src/JBSucker.sol (hunk 3 above)
Comment (xBA5ED) [reply src/JBSucker.sol:119]: We might want to store them some other way thats more optimized in the future, doing it this way forces sucker implementations to use the getter method which makes changes in the future easier.

**Pull request merged**